node/lib/util/stash_util.js

```diff
@@ -178,34 +247,42 @@ report this.  Continuing stash anyway.`);
 
         if (null === wd) {
             // closed submodule
-            if (sub.commit.sha === sub.index.sha) {
-                // ... with no staged changes
-                return;                                               // RETURN
-            }
-            // This is a case that regular git stash doesn't really have
-            // to handle. In a normal stash commit, the tree points
-            // to the working directory tree, but here, there is no working
-            // directory.  But if there were, we would want to have
-            // this commit checked out.
-
-            const subRepo = yield SubmoduleUtil.getRepo(repo, name);
+            if (sub.index === null || sub.index.sha === null) {
+                // deleted submodule
+                stashId = MAGIC_DELETED_SHA;
```

Comment:
does this make a special stash, that would be different than a user made stash?

Reply:
It's different in the meta repo, but not in the submodule repo.  Since regular stashes in the meta repo are not really a thing that does anything, I think this is OK.